docs/fxa-engineering/fxa-webchannel-protocol.md

```diff
@@ -0,0 +1,324 @@
+#### fxaccounts:fxa_status
+
+Sent on startup to fetch the FxA state from supporting browsers.
+Used to get the browser's currently signed in user as well as browser capabilities.
+
+##### request data
+
+```js
+{
+  isPairing: <boolean>
+  service: <string>
+}
+```
+
+- `isPairing`: whether or not accounts.firefox.com is executing a pairing flow
+- `service`: the service name or OAuth client-id that is requesting authentication
+
+##### response data
+
+```js
+{
+  capabilities: <capabilities>
+  signedInUser: <signed in user> || null
+}
+```
+
+- `capabilities` - A list of capabilities supported by the browser. See [capabilities](#capabilities).
+- `signedInUser` - the user currently signed into Sync on the browser. See [signedInUser](#signedInUser).
+
+##### support
+
+Fx Desktop 55+
+
+#### fxaccounts:loaded
+
+Sent on startup. Sent after the first screen is visible. No data is sent. No response is expected.
+
+#### fxaccounts:can_link_account
+
+Sent on signin or signup after a user has entered their credentials and submit the form. A response with an `ok` field is expected.
+
+##### request data
+
+```js
+{
+  email: <email>
+}
+```
+
+##### response data
+
+```js
+{
+  ok: true | false;
+}
+```
+
+#### fxaccounts:login
+
+Sent when a user successfully authenticates with Firefox Accounts and sync can begin. This includes after a user signs up but before they verify their email address, after an account unlock, and after a password reset. No response is expected.
+
+##### data
+
+See [Login Data](#logindata).
+
+#### fxaccounts:oauth_login
+
+Sent when a user successfully authenticates via a WebChannel OAuth flow.
+
+##### data
+
+```
+{
+  "code": "02f3cfea84ac4c143662b38d6c7f0c82c6f91eb041befc7cecda446b1b4887c1",
+  "state": "vHao1p6OizzwReCkQMSpZA",
+  "redirect": "urn:ietf:wg:oauth:2.0:oob:oauth-redirect-webchannel",
+  "action": "signin",
+  "declinedSyncEngines": ["history"],
+  "offeredSyncEngines": ["history"]
+}
+```
+
+##### support
+
+Supported by `oauth_webchannel_v1` context.
+
+#### fxaccounts:verified
+
+The user has successfully verified their email address.
+
+##### data
+
+See [Login Data](#logindata).
+
+##### support
+
+#### fxaccounts:delete_account
+
+Sent after the user successfully deletes their account. No response is expected.
+Expected by Fennec.
+
+##### data
+
+```js
+{
+  email: <email>,
+  uid: <user id>
+}
+```
+
+- `email` - User's email address.
+- `uid` - The user identifier.
+
+#### fxaccounts:delete
+
+Sent after the user successfully deletes their account. No response is expected.
+Expected by Fx Desktop.
+
+##### data
+
+```js
+{
+  email: <email>,
+  uid: <user id>
+}
+```
+
+- `email` - User's email address.
+- `uid` - The user identifier.
+
+#### fxaccount:change_password
+
+Sent after the user successfully changes their password. No response is expected. A change password causes a user's `unwrapBKey` to change, all data that is sent with `fxaccounts:login` is sent with `fxaccounts:change_password`.
```

Comment:
> A change password causes a user's `unwrapBKey` to change,
> all data that is sent with `fxaccounts:login` is sent with `fxaccounts:change_password`.

Something about this doesn't quite parse right for me, but I also wonder what the recipient is supposed to *do* with the information about `unwrapBKey` changing. I would not expect them to have to re-fetch keys, so maybe we can just not mention `unwrapBKey` here.